Detectors/TPC/simulation/include/TPCSimulation/GBTFrameContainer.h

```diff
@@ -0,0 +1,233 @@
+    void reset();
+
+    /// Get the size of the container
+    /// @return Size of GBT frame container
+    int getSize() { return mGBTFrames.size(); };
+
+    /// Get the number of entries in the container
+    /// @return Number of entries in the GBT frame container
+    int getNentries();
+
+    /// Add copy of frame to the container
+    /// @param frame GBT Frame
+    void addGBTFrame(GBTFrame& frame);
+
+    /// Add frame to the container
+    /// @param word3 Word 3 of GBT frame, contains bit [127: 96], [127:112] are not part of the actual frame
+    /// @param word2 Word 2 of GBT frame, contains bit [ 95: 64]
+    /// @param word1 Word 1 of GBT frame, contains bit [ 63: 32]
+    /// @param word0 Word 0 of GBT frame, contains bit [ 31:  0]
+    void addGBTFrame(unsigned& word3, unsigned& word2, unsigned& word1, unsigned& word0);
+
+    /// Add frame to the container
+    /// @param s0hw0l half-word 0 from SAMPA 0 low channel numbers 
+    /// @param s0hw1l half-word 1 from SAMPA 0 low channel numbers 
+    /// @param s0hw2l half-word 2 from SAMPA 0 low channel numbers 
+    /// @param s0hw3l half-word 3 from SAMPA 0 low channel numbers 
+    /// @param s0hw0h half-word 0 from SAMPA 0 high channel numbers 
+    /// @param s0hw1h half-word 1 from SAMPA 0 high channel numbers 
+    /// @param s0hw2h half-word 2 from SAMPA 0 high channel numbers 
+    /// @param s0hw3h half-word 3 from SAMPA 0 high channel numbers 
+    /// @param s1hw0l half-word 0 from SAMPA 1 low channel numbers 
+    /// @param s1hw1l half-word 1 from SAMPA 1 low channel numbers 
+    /// @param s1hw2l half-word 2 from SAMPA 1 low channel numbers 
+    /// @param s1hw3l half-word 3 from SAMPA 1 low channel numbers 
+    /// @param s1hw0h half-word 0 from SAMPA 1 high channel numbers 
+    /// @param s1hw1h half-word 1 from SAMPA 1 high channel numbers 
+    /// @param s1hw2h half-word 2 from SAMPA 1 high channel numbers 
+    /// @param s1hw3h half-word 3 from SAMPA 1 high channel numbers 
+    /// @param s2hw0 half-word 0 from SAMPA 2
+    /// @param s2hw1 half-word 1 from SAMPA 2
+    /// @param s2hw2 half-word 2 from SAMPA 2
+    /// @param s2hw3 half-word 3 from SAMPA 2
+    /// @param s0adc ADC clock from SAMPA 0
+    /// @param s1adc ADC clock from SAMPA 1
+    /// @param s2adc ADC clock from SAMPA 2
+    /// @param marker additional 16 bit marker which is not part of the actual frame
+    void addGBTFrame(short& s0hw0l, short& s0hw1l, short& s0hw2l, short& s0hw3l,
+                     short& s0hw0h, short& s0hw1h, short& s0hw2h, short& s0hw3h,
+                     short& s1hw0l, short& s1hw1l, short& s1hw2l, short& s1hw3l,
+                     short& s1hw0h, short& s1hw1h, short& s1hw2h, short& s1hw3h,
+                     short& s2hw0,  short& s2hw1,  short& s2hw2,  short& s2hw3, 
+                     short& s0adc,  short& s1adc,  short& s2adc,  unsigned marker = 0);
+
+    template<typename... Args> void addGBTFrame(Args&&... args);
```

Comment:
this function is missing implementation (and should probably be implemented in the header).